TBG.Business/SingleEliminationTournament.cs

```diff
@@ -9,14 +9,124 @@ namespace TBG.Business
 {
     public class SingleEliminationTournament : ITournament
     {
-        public int TournamentId { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int UserId { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public string TournamentName { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public decimal EntryFee { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public double TotalPrizePool { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public int TournamentTypeId { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public List<ITeam> Participants { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        public List<ITournamentPrize> Prizes { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
-        List<ITournamentEntry> ITournament.Participants { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public int TournamentId { get; set; }
+        public int UserId { get; set; }
+        public string TournamentName { get; set; }
+        public decimal EntryFee { get; set; }
+        public double TotalPrizePool { get; set; }
+        public int TournamentTypeId { get; set; }
+        public List<ITournamentEntry> Participants { get; set; }
+        public List<ITournamentPrize> Prizes { get; set; }
+        public List<IRound> Rounds { get; set; }
+
+        public SingleEliminationTournament()
+        {
+            Participants = new List<ITournamentEntry>();
+            Prizes = new List<ITournamentPrize>();
+            Rounds = new List<IRound>();
+        }
+
+        public bool BuildTournament()
+        {
+            Queue<ITournamentEntry> teamQueue = new Queue<ITournamentEntry>();
+            foreach (var team in Participants)
+            {
+                teamQueue.Enqueue(team);
+            }
+
+            for (int i = 1; i <= CalculateRoundTotal(Participants.Count); i++)
```

Comment:
I think it'd be easier to read here if CalculateRoundTotal(Participants.Count) is put in a variable called "totalRounds" before the loop.

Reply:
Yup, that's not a problem.  I will make that change.